website_customer_also_purchased/model/product.py

```diff
@@ -23,59 +23,46 @@
 #
 ##############################################################################
 
-from openerp.osv import osv
-from openerp.osv import fields
-from openerp import SUPERUSER_ID
+from openerp import api, models
 
 
-class ProductTemplate(osv.osv):
+class ProductTemplate(models.Model):
     _inherit = 'product.template'
 
-    def _get_purchased(self, cr, uid, ids, field_names, arg=None,
-                       context=None):
+    def _best_sell_sort(self, product_tmp, offset=0, max_product_qty=6):
+        variants_ids = '(%s)' % ', '.join(
+            [str(v.id) for v in product_tmp.product_variant_ids])
+        if variants_ids and variants_ids != '()':
+            query = "\
+                SELECT p.product_tmpl_id from sale_order_line sol \
+                join sale_order_line sol2 on sol.order_id=sol2.order_id \
+                join product_product p on sol2.product_id = p.id \
+                join product_template p_tmpl on p.product_tmpl_id=p_tmpl.id \
+                where sol.product_id in {0} and p.product_tmpl_id != {1} \
+                and p_tmpl.website_published=True \
+                group by p.product_tmpl_id \
+                order by count(sol2.product_id) desc LIMIT {2} OFFSET {3};\
+                    ".format(variants_ids, product_tmp.id,
```

Comment:
Please give a better format to SQL.

Something like this (in terms of format and LINTS.).
https://github.com/odoo/odoo/blob/9.0/addons/account/models/account.py#L60